modules/renter/siafile/siafile.go

```diff
@@ -0,0 +1,181 @@
+package siafile
+
+import (
+	"bytes"
+	"encoding/base32"
+	"encoding/binary"
+	"fmt"
+	"sync"
+
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+	"github.com/NebulousLabs/fastrand"
+
+	"github.com/NebulousLabs/Sia/crypto"
+)
+
+type (
+	// SiaFile is the disk format for files uploaded to the Sia network.  It
+	// contains all the necessary information to recover a file from its hosts and
+	// allows for easy constant-time updates of the file without having to read or
+	// write the whole file.
+	SiaFile struct {
+		// metadata is the mostly static metadata of a SiaFile. The reserved
+		// size of the metadata on disk should always be a multiple of 4kib.
+		// The metadata is also the only part of the file that is JSON encoded
+		// and can therefore be easily extended.
+		metadata Metadata
```

Comment:
@lukechampine @ChrisSchinnerl what do you think about embedding the `Metadata` field into the SiaFile struct, that way every method and field of the metadata can be accessed directly?

Reply:
The reason why I didn't do that is because the `Metadata` will be JSON encoded. That way we can just pass the `Metadata` field directly to the encoder. Otherwise we need a separate struct for that.